contrib/twmb/franz-go/franz/headers.go

```diff
@@ -0,0 +1,57 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package franz
+
+import (
+	"github.com/twmb/franz-go/pkg/kgo"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// A RecordHeadersCarrier injects and extracts traces from kgo.Record headers.
+type RecordHeadersCarrier struct {
+	headers []kgo.RecordHeader
+}
+
+var _ interface {
+	tracer.TextMapReader
+	tracer.TextMapWriter
+} = (*RecordHeadersCarrier)(nil)
+
+// ForeachKey iterates over every header.
+func (c RecordHeadersCarrier) ForeachKey(handler func(key, val string) error) error {
```

Comment:
Any reason why `ForeachKey` is called on `RecordHeadersCarrier` instead of a pointer to it (`*RecordHeadersCarrier`)?